src/FindApprenticeshipTraining/SFA.DAS.FindApprenticeshipTraining/Application/Courses/Queries/GetCourseProviders/GetTrainingCourseProvidersQueryHandler.cs

```diff
@@ -1,78 +1,70 @@
+namespace SFA.DAS.FindApprenticeshipTraining.Application.Courses.Queries.GetCourseProviders;
+
+public class GetTrainingCourseProvidersQueryHandler(IRoatpCourseManagementApiClient<RoatpV2ApiConfiguration> _roatpCourseManagementApiClient, ICachedLocationLookupService _cachedLocationLookupService, ICoursesApiClient<CoursesApiConfiguration> coursesApiClient) : IRequestHandler<GetCourseProvidersQuery, GetCourseProvidersResponse>
 {
-    public class GetTrainingCourseProvidersQueryHandler(IRoatpCourseManagementApiClient<RoatpV2ApiConfiguration> _roatpCourseManagementApiClient, ILocationLookupService locationLookupService, ICoursesApiClient<CoursesApiConfiguration> coursesApiClient) : IRequestHandler<GetCourseProvidersQuery, GetCourseProvidersResponse>
+    public async Task<GetCourseProvidersResponse> Handle(GetCourseProvidersQuery request, CancellationToken cancellationToken)
     {
+        LocationItem locationItem = null;
 
-        public async Task<GetCourseProvidersResponse> Handle(GetCourseProvidersQuery request,
-            CancellationToken cancellationToken)
+        if (!string.IsNullOrWhiteSpace(request.Location))
         {
-            decimal? latitude = null;
-            decimal? longitude = null;
+            locationItem = await _cachedLocationLookupService.GetCachedLocationInformation(request.Location);
 
-            if (request.Location != null)
+            if (locationItem is null)
             {
-                var location = await locationLookupService.GetLocationInformation(request.Location, 0, 0);
-
-                if (location == null)
-                {
-                    GetStandardsListItem standard =
-                        await coursesApiClient.Get<GetStandardsListItem>(new GetStandardRequest(request.Id));
+                GetStandardsListItem standard = await coursesApiClient.Get<GetStandardsListItem>(new GetStandardRequest(request.Id));
 
-                    var standardName = standard != null ? $"{standard.Title} (level {standard.Level})" : "";
-
-                    var responseToSendBack = new GetCourseProvidersResponse
-                    {
-                        PageSize = 10,
-                        Page = 1,
-                        LarsCode = request.Id,
-                        Providers = new List<ProviderData>(),
-                        QarPeriod = string.Empty,
-                        ReviewPeriod = string.Empty,
-                        StandardName = standardName,
-                        TotalCount = 0,
-                        TotalPages = 0
-                    };
-
-                    return responseToSendBack;
-                }
+                var standardName = standard != null ? $"{standard.Title} (level {standard.Level})" : string.Empty;
 
-                latitude = (decimal)location.GeoPoint[0];
-                longitude = (decimal)location.GeoPoint[1];
+                return new GetCourseProvidersResponse
+                {
+                    PageSize = 10,
```

Comment:
can you update the magic numbers for pagesize and page to constants?  And update the coverage to use those too